code/edge/src/main/java/com/adobe/marketing/mobile/EdgeState.java

```diff
@@ -55,11 +57,21 @@ class EdgeState {
 	 * <p>
 	 * Loads any persisted Edge properties and creates an initial shared state.
 	 *
-	 * @param registeredExtensionsWithHub the extensions registered with the Event Hub
+	 * @return true if bootup is complete
 	 */
-	void bootupIfNeeded(final Map<String, Object> registeredExtensionsWithHub) {
+	boolean bootupIfNeeded() {
 		if (hasBooted) {
-			return;
+			return true;
+		}
+
+		// Get Hub's shared state needed to check if Identity and Consent are registered
+		// If not set, return false and attempt bootup at next event
+		final SharedStateResult eventHubStateResult = sharedStateCallback.getSharedState(
```

Comment:
Does the actual check on whether Consent and Identity are registered happen elsewhere?

Reply:
I'm not sure why I said Identity here. The hub's state is needed to build ImplementationDetails using the Core version and wrapper type, and setting the default Consent if Consent is not registered. Both of those a checked in the synchronized block, in `ImplementationDetails.fromEventHubState` and `updateCollectConsentIfNotRegistered`.
I'll update the comment to make it more clear.